packages/webui/components/types/resources.ts

```diff
@@ -0,0 +1,11 @@
+export type ResourceMetadata = {
+    uri: string;
+    name?: string;
+    description?: string;
+    mimeType?: string;
+    source: 'mcp' | 'plugin' | 'custom';
+    serverName?: string;
+    size?: number;
+    lastModified?: string | Date;
+    metadata?: Record<string, unknown>;
+};
```

Comment:
import the type from core, libraries are properly separated now

limit duplication as much as possible